activerecord/lib/active_record/associations.rb

```diff
@@ -1251,6 +1274,22 @@ module ClassMethods
         #   persisted new records placed at the end.
         #   When set to +:nested_attributes_order+, the index is based on the record order received by
         #   nested attributes setter, when accepts_nested_attributes_for is used.
+        # [:before_add]
+        #   Defines a callback that gets triggered before an object is added to the association collection.
+        #
+        #   Read more about Associations::ClassMethods@Association+callbacks
```

Comment:
Do we really need this after every option?

Reply:
For documentation, I favor redundancy and discoverability over brevity. Having said that, do you have an idea on how to group the options in a way that would make a single mention suitable?

Reply:
![Screenshot from 2024-06-20 19-05-39](https://github.com/rails/rails/assets/277819/4b257f37-38f5-4477-9b7a-04afc1e7c266)

This is unnecessary redundancy IMO. I don't think options should be grouped if they are separate keys, I would just remove the link from each option.

I also think there is a bit of "which of these options are unique to this method?" and "which of these options apply to all (or most) of the methods here?", and are those commonly shared options documented properly in the enclosing module/class?

My feeling is it is a bit of the latter, and I would prefer not to explicitly list every shared option across every method, but to link to "all supported options for these set of methods" in the context of the module.

Reply:
I think I agree with @zzak. We also don't link to Validations or StrictLoading everytime they are mentioned.

If we do add a link, I think it should be inline and link to the Association Callbacks section higher up.
https://2362cf21.rails-docs-preview.pages.dev/api/classes/ActiveRecord/Associations/ClassMethods#module-ActiveRecord::Associations::ClassMethods-label-Association+callbacks
```
Defines a [association callback](url_to_association_callbacks_section) that gets triggered before an object is added to the association collection.
```